False positive for `positional-only-arguments-expected`

### Bug description

a.py:
```python
def function_with_kwargs(apple, banana="Yellow banana", /, **kwargs):
    """
    Calling this function with the `banana` keyword should not emit
    `positional-only-arguments-expected` since it is added to `**kwargs`.

    >>> function_with_kwargs("Red apple", banana="Green banana")
    >>> "Red apple"
    >>> "Yellow banana"
    >>> {"banana": "Green banana"}
    """
    print(apple)
    print(banana)
    print(kwargs)
```

### Configuration

_No response_

### Command used

```shell
pylint a.py
```


### Pylint output

```shell
a.py:15:0: E3102: `function_with_kwargs()` got some positional-only arguments passed as keyword arguments: 'banana' (positional-only-arguments-expected)
```


### Expected behavior

No error emitted in this case as the argument passed to the function is part of `**kwargs`.
The purpose of this checker is to indicate when an error would occur at runtime, for example:
`TypeError: function_with_kwargs() got some positional-only arguments passed as keyword arguments: 'banana'`.

However, in this case, there is no error running this Python code.

### Pylint version

```shell
pylint 3.0.0b1
astroid 2.16.0dev0
Python 3.10.4
```


### OS / Environment

_No response_

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/8555.false_positive
+++ b/doc/whatsnew/fragments/8555.false_positive
@@ -1,0 +1,3 @@
+Fix false positive for ``positional-only-arguments-expected`` when a function contains both a positional-only parameter that has a default value, and ``**kwargs``.
+
+Closes #8555

--- a/pylint/checkers/method_args.py
+++ b/pylint/checkers/method_args.py
@@ -111,6 +111,8 @@ class MethodArgsChecker(BaseChecker):
             and inferred_func.args.posonlyargs
         ):
             return
+        if inferred_func.args.kwarg:
+            return
         pos_args = [a.name for a in inferred_func.args.posonlyargs]
         kws = [k.arg for k in node.keywords if k.arg in pos_args]
         if not kws:

--- a/tests/functional/p/positional_only_arguments_expected.py
+++ b/tests/functional/p/positional_only_arguments_expected.py
@@ -16,3 +16,21 @@ cake.nihon(1, 2, i=3)  # [positional-only-arguments-expected]
 cake.nihon(1, r=2, i=3)  # [positional-only-arguments-expected]
 cake.nihon(a=1, r=2, i=3)  # [positional-only-arguments-expected]
 cake.nihon(1, r=2, i=3, cheese=True)  # [positional-only-arguments-expected]
+
+
+def function_with_kwargs(apple, banana="Yellow banana", /, **kwargs):
+    """
+    Calling this function with the `banana` keyword should not emit
+    `positional-only-arguments-expected` since it is added to `**kwargs`.
+
+    >>> function_with_kwargs("Red apple", banana="Green banana")
+    >>> "Red apple"
+    >>> "Yellow banana"
+    >>> {"banana": "Green banana"}
+    """
+    print(apple)
+    print(banana)
+    print(kwargs)
+
+
+function_with_kwargs("Red apple", banana="Green banana")

